Mathlib/RingTheory/DedekindDomain/IntegralClosure.lean

```diff
@@ -246,9 +252,44 @@ instance integralClosure.isDedekindDomain_fractionRing [IsDedekindDomain A] :
     IsDedekindDomain (integralClosure A L) :=
   integralClosure.isDedekindDomain A (FractionRing A) L
 
+open nonZeroDivisors Algebra IsLocalization IsFractionRing IsScalarTower in
+instance [NoZeroSMulDivisors A C] [Module.Finite A C] :
+    IsLocalization (Algebra.algebraMapSubmonoid C A⁰) (FractionRing C) := by
+  refine ⟨fun ⟨s, r, hr, hrs⟩ ↦ ?_, fun z ↦ ?_, fun {x y} h ↦ ⟨1, ?_⟩⟩
+  · simp only [isUnit_iff_ne_zero, ne_eq, FaithfulSMul.algebraMap_eq_zero_iff]
+    intro hs
+    simp only [hs, FaithfulSMul.algebraMap_eq_zero_iff] at hrs
+    simp [hrs] at hr
+  · let B := Localization (algebraMapSubmonoid C A⁰)
+    have : IsDomain B := isDomain_localization <|
+      map_le_nonZeroDivisors_of_injective _ (FaithfulSMul.algebraMap_injective _ _) rfl.le
+    have : IsScalarTower A (FractionRing A) B :=
+      instIsScalarTowerLocalizationAlgebraMapSubmonoid A⁰ C
+    have : Module.Finite (FractionRing A) B := Module.finite_of_isLocalization A C _ B A⁰
```

Comment:
Can these be instances as well?
Or rather, are they not `infer_instance` yet?

Reply:
Added the instance.